crates/forge/src/scarb.rs

```diff
@@ -50,16 +72,63 @@ pub struct ForgeConfig {
     pub fuzzer_seed: Option<u64>,
 
     #[serde(default)]
-    pub fork: Vec<ForkTarget>,
+    pub fork: Vec<RawForkTarget>,
 }
 
 #[derive(Deserialize, Debug, PartialEq, Default, Clone)]
-pub struct ForkTarget {
+pub struct RawForkTarget {
     pub name: String,
     pub url: String,
     pub block_id: HashMap<String, String>,
 }
 
+impl TryFrom<RawForgeConfig> for ForgeConfig {
+    type Error = anyhow::Error;
+
+    fn try_from(value: RawForgeConfig) -> std::result::Result<Self, Self::Error> {
+        let mut fork_targets = vec![];
+
+        for raw_fork_target in value.fork {
+            let block_id: Vec<BlockId> = raw_fork_target
+                .block_id
+                .iter()
+                .map(|(id_type, value)| match id_type.as_str() {
+                    "number" => BlockId::Number(value.parse().unwrap()),
+                    "hash" => BlockId::Hash(value.to_field_element()),
+                    "tag" => match value.as_str() {
+                        "Latest" => BlockId::Tag(BlockTag::Latest),
+                        "Pending" => BlockId::Tag(BlockTag::Pending),
+                        _ => unreachable!(),
+                    },
+                    _ => unreachable!(),
+                })
+                .collect();
+
+            let [block_id] = block_id[..] else {
+                unreachable!()
+            };
+
+            fork_targets.push(ForkTarget {
+                name: raw_fork_target.name,
+                url: Url::parse(&raw_fork_target.url).map_err(|parse_error| {
+                    anyhow!(
+                        "Could not parse the \"{}\" URL from Scarb.toml: {parse_error}",
+                        raw_fork_target.url
```

Comment:
Nit but we usually use constructs like `url = {}` in these errors instead of `""` or `:`.